junit-jupiter-api/src/main/kotlin/org/junit/jupiter/api/Assertions.kt

```diff
@@ -147,6 +147,14 @@ inline fun <reified T : Throwable> assertThrows(noinline message: () -> String,
     }, Supplier(message))
 }
 
+@PublishedApi
```

Comment:
https://stackoverflow.com/a/41905907

Reply:
Again, possibly dumb question here, but why do we need this method to be `inline`?

Reply:
Also, could you please move this method below the other ones calling it? :)

Reply:
> Again, possibly dumb question here, but why do we need this method to be `inline`?

Because this function is ultimately the one responsible for calling the provided `executable`. We want to run the `executable` in the same context as the callsite of `assertDoesNotThrow` (so that it can suspend, etc.), so we need to make sure the full call stack from `assertDoesNotThrow` until here uses inlined functions. If we were to remove the `inline` modifier on this function, we'll get a compiler error in all of the `assertDoesNotThrow` functions saying that we're violating the inlineability <sup>(I totally just made that word up)</sup> of the `executable` argument, and that we need to change the function signature to `assertDoesNotThrow(noinline executable: () -> R)`. Doing so breaks the tests that now calls a suspending function, because the callback is no longer inlined.

Reply:
I see! Thanks for the explanation 😄